pycaret/internal/ts_tests.py

```diff
@@ -0,0 +1,227 @@
+    t_values = decompose_result.trend.fillna(method='bfill').fillna(method='ffill')
+
+    trend = find_trend(t_values)
+    if plots:
+        f, ax = plt.subplots(nrows=2, ncols=1)
+
+        # trend plot
+        ax[0].plot(ts.index, t_values)
+        ax[0].plot(ts.index, trend * np.arange(1, len(t_values) + 1))
+        ax[0].legend(['Trend from stl', 'Linear trend'])
+        ax[0].set_title('Trend plot', size=12)
+
+        # seasonal plot
+        df.plot.bar(ax=ax[1], rot=0)
+        plt.hlines(1, -1, len(df.index), linestyles='--', colors='black', alpha=0.5)
+        ax[1].legend(['Significance level', 'Seasonal value'])
+        ax[1].set_title('Seasonal plot', size=12)
+
+        f.tight_layout()
+
+    if abs(trend) >= 1:
+        trend_nature = f"Strong - {'increasing' if trend > 0 else 'decreasing'} trend"
+    elif 0.1 < abs(trend) < 1:
+        trend_nature = f"Weak - {'increasing' if trend > 0 else 'decreasing'} trend"
+    else:
+        trend_nature = 'Trend absent'
+    result = {
+        'Trend': trend_nature,
+        'Trend Slope': trend,
+        'Seasonal': 'Seasonal in nature' if is_seasonal else 'Seasonality absent',
+        'Seasonal Period': seasonal_period,
+    }
+    return result
+
+
+def statistic_tests(
+    ts: Union[pd.DataFrame, pd.Series],
+    plots=False,
+    **kwargs
+):
+    from scipy.stats import shapiro, norm
+
+    normality_pvalue = shapiro(ts.values.squeeze())[1]
+
+    if normality_pvalue > 0.05:
+        normality = 'Gaussian'
+    else:
+        normality = 'Not gaussian'
+
+    distinct_counts = dict(ts.iloc[:, 0].value_counts(normalize=True))
+    mean = ts.mean()[0]
+    std = ts.std()[0]
+    result = {
+        'Mean': mean,
+        'Median': ts.median()[0],
+        'Standard Deviation': std,
+        'Variance': ts.var()[0],
+        'Kurtosis': ts.kurt()[0],
+        'Skewness': ts.skew()[0],
+        'Count of distinct values': len(distinct_counts),
+        '_Data distribution': {
+            key: f'{round(val * 100)}%' for key, val in
+            zip(list(distinct_counts.keys())[:3], list(distinct_counts.values())[:3])
+        },
+        'Length': len(ts),
+        'Shapiro normality test': normality,
+        '_Normality pvalue': round(normality_pvalue, 3)
+    }
+    if plots:
+        sns.distplot(ts, fit=norm)
+        plt.show()
+    return result
+
+
+def correlation_tests(
+    ts: Union[pd.DataFrame, pd.Series],
+    plots=False,
+    **kwargs
+):
+    from statsmodels.tsa.stattools import acf, pacf
+
+    acf_x, ci = acf(ts, alpha=0.05, nlags=20)
+    lower = ci[:, 0] - acf_x
+    upper = ci[:, 1] - acf_x
+    acf_x = np.round(((acf_x >= upper) + (acf_x <= lower)) * acf_x, 2)
+    acf_map = {val: idx for idx, val in enumerate(acf_x)}
+    acf_values = list(filter(lambda x: abs(x) > 0, acf_x))
+    acf_lags = list(map(lambda x: acf_map[x], acf_values))
+
+    pacf_x, ci = pacf(ts, alpha=0.05, nlags=20)
+    lower = ci[:, 0] - pacf_x
```

Comment:
Since you are comparing pacf to upper and lower ci below, do we need to subtract it here?

```python
pacf_x >= upper) + (pacf_x <= lower)
```